lib/segment/src/index/field_index/stat_tools.rs

```diff
@@ -0,0 +1,145 @@
+use std::f64::consts::{E, PI};
+
+/// This function estimates how many real points were selected with the filter.
+/// It is assumed that each real point has, on average, X values in correspondence.
+/// As a response to the execution of the query it is possible to establish only the number of matched associated values.
+///
+///
+/// # Arguments
+///
+/// * `total_points` - total number of the unique points in the whole collection
+/// * `total_values` - total number of payload values in the collection
+/// * `selected_values_count` - amount of values selected during the query
+///
+/// # Result
+///
+/// Expected amount of unique points contained in selected values
+/// The result might overflow at some corner cases
+///   so it is better to limit its value with min and max
+///
+pub fn estimate_multi_value_selection_cardinality(
+    total_points: usize,
+    total_values: usize,
+    selected_values_count: usize,
+) -> f64 {
+    // Value >= 1.0
+    let values_per_point = total_values as f64 / total_points as f64;
+    // Probability to select each unique value
+    let prob_select = 1. - prob_not_select(total_values, values_per_point, selected_values_count);
+    prob_select * total_points as f64
+}
+
+/// Fast approximate computation of $ln(n!)$
+/// See: <https://en.wikipedia.org/wiki/Stirling%27s_approximation>
+fn approx_fact_log(n: f64) -> f64 {
+    if n < 1.0 {
+        return 1.0; // By definition
+    }
+    (2. * PI * n).sqrt().ln() + n * (n / E).ln()
```

Comment:
the std lib defines also `TAU` that we could use here

```rust
    /// The full circle constant (τ)
    ///
    /// Equal to 2π.
    #[stable(feature = "tau_constant", since = "1.47.0")]
    pub const TAU: f64 = 6.28318530717958647692528676655900577_f64;
```